internal/utils/flags/flags.go

```diff
@@ -0,0 +1,122 @@
+package flags
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	//TypeString is the type of strings
+	TypeString = "string"
+)
+
+// enumSetValue is modified copy of stringSliceValue from
+// the cobra pflags package
+type enumSetValue struct {
+	value          *[]string
+	changed        bool
+	validValuesSet map[string]bool
+}
+
+// NewEnumSetValue creates a stringSliceValue that validates inputs and prevents duplicates
+func NewEnumSetValue(p *[]string, val []string, validValues []string) pflag.Value {
+	esv := new(enumSetValue)
+	esv.value = p
+	*esv.value = val
+	esv.validValuesSet = make(map[string]bool)
+	for _, validValue := range validValues {
+		esv.validValuesSet[validValue] = true
+	}
+	return esv
+}
+
+func (esv *enumSetValue) Set(val string) error {
+	v, err := readAsCSV(val)
+	if err != nil {
+		return err
+	}
+	if !esv.changed {
+		*esv.value = v
+	} else {
+		*esv.value = append(*esv.value, v...)
+	}
+	esv.changed = true
+	return esv.validateAndRemoveDuplicates()
+}
+
+func (esv *enumSetValue) Type() string {
+	return "stringSlice"
+}
+
+func (esv *enumSetValue) String() string {
+	str, _ := writeAsCSV(*esv.value)
+	return "[" + str + "]"
+}
+
+func (esv *enumSetValue) Append(val string) error {
+	*esv.value = append(*esv.value, val)
+	return esv.validateAndRemoveDuplicates()
+}
+
+func (esv *enumSetValue) Replace(val []string) error {
+	*esv.value = val
+	return esv.validateAndRemoveDuplicates()
+}
+
+func (esv *enumSetValue) GetSlice() []string {
+	return *esv.value
+}
+
+func (esv *enumSetValue) validateAndRemoveDuplicates() error {
```

Comment:
[question] is this necessary?  couldn't we also keep a `valueSet map[string]struct{}` field on `enumSetValue` populated as we call `Set`/`Append`/`Replace` so we can avoid building this set each and every time `validateAndRemoveDuplicates` is called?

Reply:
Sure, I didn't think performance was at all a consideration for flag parsing, so I just did it this way cuz it reads simpler.  I'll add it to the struct then